entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local filetype=$1
+  shift
+
+  matching_files=()
+
+  for file in "$@"; do
+    filename=$(basename "$file")
+    
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
```

Comment:
We should not overload vars and function names.  Really this is something like:
```suggestion
function check_linter_exit_code() {
```

Reply:
Done